forloop_modules/function_handlers/database_handlers.py

```diff
@@ -704,20 +778,45 @@ def execute(self, node_detail_form):
 
         self.direct_execute(db_name, db_table_name, column_name, operator, value)
 
-    def direct_execute(self, db_name,  dbtable_name, column_name, operator, value):
-        db_table = get_db_table_from_db(table_name=dbtable_name, db_name=db_name)
-        db_instance = db_table.db1
+    def direct_execute(self, db_name,  db_table_name, column_name, operator, value):
+        project_databases = get_all_databases_in_project()
+        db_dict = filter_database_by_name_from_all_project_databases(project_databases=project_databases, db_name=db_name)
+        
+        if db_dict is None:
+            # User selects from stored DBs so this shouldn't happen. If this is raised, these is an issue in code probably.
+            raise Exception(f'{self.icon_type}: No DB named {db_name} found in project DBs.')
+        
+        redis_key = create_redis_key_for_project_db_private_key(project_uid=aet.project_uid)
+        private_key_base64 = kv_redis.get(redis_key)
+        
+        if private_key_base64 is not None:
+            private_key = convert_base64_private_key_to_rsa_private_key(private_key_base64=private_key_base64)
+            
+            encrypted_password = db_dict["password"]
+            decrypted_password = decrypt_text(text=encrypted_password, private_key=private_key)
+            
+            db_dict["password"] = decrypted_password
+            
+            db_details = dbc.create_db_details_from_database_dict(db_dict=db_dict)
+            db_connection = dbc.DbConnection(db_details=db_details)
+            is_connected = db_connection.test_database_connection()
+        
+            if is_connected:
+                db_instance = db_connection.db_instance
+                db_table = db_connection.get_db_table(db_table_name)
+                if db_table is None:
+                    return
         
-        value = parse_float_db(db_instance=db_instance, value=value)
+                value = parse_float_db(db_instance=db_instance, value=value)
 
-        if type(db_instance) is dh.MongoDb:
-            where_statement = get_condition_mongo(column_name, value, operator)
-        else:
-            where_statement = f"{column_name}{operator}{value}"
-            if where_statement == "*='*'": #delete all data from table -> column_name = ["*"], operator = "=", value = '*'
-                where_statement = None
+                if type(db_instance) is dh.MongoDb:
+                    where_statement = get_condition_mongo(column_name, value, operator)
+                else:
+                    where_statement = f"{column_name}{operator}{value}"
+                    if where_statement == "*='*'": #delete all data from table -> column_name = ["*"], operator = "=", value = '*'
+                        where_statement = None
```

Comment:
this seems slightly weird, shouldnt dbhydra cover this example? Why do you need it here

Reply:
If its because we are sending empty where_statement from the item_detail_form, let's rather put if condition on the empty inputs

Reply:
Don't know tbh. I kept the insides as they were since they work. I can take a look at it though.